### DISCUSSION TITLE
Provide an easy / "one click" / "one script to run" way to install reasonably configured language support for a variety of languages?

### DISCUSSION THREAD

**Discussion by jerabaul29** [helix-editor/helix#6999] [Ideas]
I am aware that there is documentation about installing LSP support for languages in the documentation at https://docs.helix-editor.com/languages.html . My feeling is that for very many users, having a "reasonably default-configured" LSP support for the language(s) they use is enough, and it would be nice to be able to get it in one click / one command / one script run without the need to set things up oneself. This is one of the strengths of VSC: you can just find an extension, click on it, and it is installed and ready to use.

Do you think it would be possible to provide a similar experience (and how?)? I understand that this will involve both installing the LSP server tool (which will be host / OS dependent), and editing the config linking to the LSP bin (which will also be host / OS dependent), but still wondering if this could be provided as an easy to apply method that requires no user fiddling (for example as a simple script to run / command to run) for a few of the most common systems (Debian and derivatives, Windows, Mac), and / or for systems that support it, as an easy to install extension for flatpak or similar (for example, using something similar to what is described by @blt-r in https://github.com/helix-editor/helix/issues/6993#issuecomment-1538950314_ )?

**Comment by kirawi** [comment]
Duplicate of https://github.com/helix-editor/helix/issues/2152

**Comment by David-Else** [comment]
It could be done as a bash script that detects your OS and then has some kind of menu system to choose what language servers you want to install and uninstall. The problem is it would be a nightmare to maintain, and not worth it in my opinion. It could use the `languages.toml` file as a reference.

Another idea, build a massive Ansible script that detects your OS and installs the server you want. It could be shared with Neovim users and get many more contributors than just Helix users. It would of course require installing Ansible, that might defeat the point for many people.

Instructions on how to install the language servers are very easy to follow.

>  it would be nice to be able to get it in one click / one command / one script run without the need to set things up oneself

They are nearly all just that, most Linux distributions have NPM etc ready to use.

I did offer to re-write [the WIKI](https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers) into the main documentation, but it was not not something that was wanted as it was a big maintenance burden. There is still a lot of work to do on that WIKI document, it needs to be more uniform.

I used the Neovim plugins that did this task, but uninstalled them in the end as they became very annoying and prone to mistakes.

**Comment by willfaught** [comment]
Perhaps there could be support for common package managers:

```sh
$ hx --packager brew --install c,cpp,cmake,make # runs brew install clangd cmake-language-server
$ hx --packager choco --install-sysdev          # runs choco install clangd cmake-language-server
```

**Reply by kirawi** [reply]
That is better suited for plugins. Helix doesn't want to handle stuff like that. Besides, it's not much work to check the wiki and download the associated LSP.
